src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/Internal/Tables.cs

```diff
@@ -1343,6 +1384,47 @@ internal int GetPropertyListStartFor(int rowId)
             int rowOffset = (rowId - 1) * this.RowSize;
             return this.Block.PeekReference(rowOffset + _PropertyListOffset, _IsPropertyRefSizeSmall);
         }
+
+        internal TypeDefinitionHandle FindTypeContainingProperty(int propertyRowId, int numberOfProperties)
+        {
+            int numOfRows = this.NumberOfRows;
+            int slot = this.Block.BinarySearchForSlot(numOfRows, this.RowSize, _PropertyListOffset, (uint)propertyRowId, _IsPropertyRefSizeSmall);
+            int row = slot + 1;
+            if (row == 0)
+            {
+                return default(TypeDefinitionHandle);
+            }
+
+            if (row > numOfRows)
+            {
+                if (propertyRowId <= numberOfProperties)
+                {
+                    return GetParentType(numOfRows);
+                }
+
+                return default(TypeDefinitionHandle);
+            }
+
+            int value = this.GetPropertyListStartFor(row);
+            if (value == propertyRowId)
+            {
+                while (row < numOfRows)
+                {
+                    int newRow = row + 1;
+                    value = this.GetPropertyListStartFor(newRow);
+                    if (value == propertyRowId)
+                    {
+                        row = newRow;
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+            }
+
```

Comment:
I do not think this is needed for property map table. Unlike typedef table that this code was copied from, the property map should not have rows for types with no properties.

Dtto for event map

Reply:
Done.